config.c

```diff
@@ -0,0 +1,294 @@
+/**
+ * Config Reader and Parser
+ * This program will read the config for lightwm in the appdata directory
+ * 
+ * Demetry Romanowski
+ * demetryromanowski@gmail.com
+ **/ 
+#include "config.h" 
+
+#include <Windows.h> 
+#include <Shlobj.h> 
+#include <wchar.h> 
+#include <stdio.h> 
+#include <strsafe.h> 
+#include <shlwapi.h> 
+#include <assert.h> 
+
+#include "error.h" 
+#include "resource.h"
+
+#include "debug.h"
+
+#define BUFF_SIZE 65536
+
+/**
+ * Config reader global vars
+ **/
+PWSTR szConfigFilePath[MAX_PATH]; 
+char* defaultConfigData = NULL; 
+
+//Should probably create a meta structure that holds the total count for now just another global variable
+ConfigItems configItems; 
+
+/**
+ * Private prototypes here
+**/
+BOOL CreateDefaultConfigFile(HINSTANCE);
+BOOL LoadDefaultConfigResourceData(HINSTANCE);
+BOOL WriteDefaultConfigDataToFile(); 
+void trim(char* str); 
+void removeControlChars(char* str); 
+size_t GetLineCount(FILE* file);
+ 
+DWORD ReadConfigFile() 
+{ 
+	//Try to open the config file based on the path
+	FILE* configFileHandle = _wfopen(*szConfigFilePath, L"r"); 
+	
+	if(configFileHandle == NULL) 
+	{
+		SetLastError(ERROR_INVALID_HANDLE); 
+		reportWin32Error(L"Config file could not be opened"); 
+		CleanupConfigReader();
+		return ERROR_INVALID_HANDLE; 
+	}
+	
+	char line[BUFF_SIZE]; //TODO must have a more clever way of getting a line length
+		
+	configItems.configItem = (ConfigItem*)malloc(sizeof(ConfigItem) * GetLineCount(configFileHandle) + 1); 
+	configItems.configItemsCount = GetLineCount(configFileHandle) + 1; 
+
+	if(configItems.configItem == NULL) 
+	{ 
+		reportWin32Error(L"Allocation ConfigItem struct"); 
+		CleanupConfigReader(); 
+		return ERROR_NOT_ENOUGH_MEMORY;
+	}
+	
+	for(size_t lineCount = 0; fgets(line, sizeof(line), configFileHandle); lineCount++) { 
+		if(strlen(line) == 0)
```

Comment:
I always use braces, even if there is only a single line, saves debugging hard to find bugs